docs/content/concepts/authentication/workspace.md

```diff
@@ -0,0 +1,143 @@
+---
+description: >
+  How to admit users into workspaces by using custom JWT validators.
+---
+
+# Per-Workspace Authentication
+
+kcp supports a range of authentication options, but all of them are global and applicable to every workspace in a kcp system. However when integrating with external partners and services, it can be beneficial to be able to admit users into a workspace that do not necessarily have access to kcp as a whole.
+
+To enable this, kcp supports per-workspace authentication. In this model, a `WorkspaceType` configures a set of additional OIDC validators that are then used by kcp in addition to the global authentication mechanisms configured with CLI flags. Every workspace using these custom workspace types will then have these additional auth methods available.
+
+This document describes how to enable and use this feature. Please refer to [OIDC Configuration](./oidc.md) for more information about the global OIDC configuration.
+
+## Feature Gate
+
+The feature is guarded by a feature gate called `WorkspaceAuthentication`, which is disabled by default. It can be independently enabled on any front-proxy and/or any kcp shard servers, though it is recommended and intended to enable it on front-proxies only. The shard support is mainly for testing and developing.
+
+Add `--feature-gates=WorkspaceAuthentication=true` to the CLI flags on the front-proxy to enable the feature.
```

Comment:
not `--proxy-feature-gates=`?

Reply:
No, the `proxy-` prefix is only on the sharded-test-server, not on the front-proxy itself.